src/main/java/org/nuxeo/ecm/csv/CSVImporterWork.java

```diff
@@ -438,17 +463,12 @@ protected DateFormat getDateFormat() {
         return dateformat;
     }
 
-    protected boolean createOrUpdateDocument(long lineNumber, String name, String type,
-            Map<String, Serializable> properties) {
-        Path targetPath = new Path(parentPath).append(name);
-        name = targetPath.lastSegment();
-        String newParentPath = targetPath.removeLastSegments(1).toString();
-        DocumentRef docRef = new PathRef(targetPath.toString());
-        if (options.getCSVImporterDocumentFactory().exists(session, newParentPath, name, type, properties)) {
-            return updateDocument(lineNumber, docRef, properties);
-        } else {
-            return createDocument(lineNumber, newParentPath, name, type, properties);
+    protected boolean isTypeColumnMissing(Map<String, Integer> header) {
+        // transient field so may become null
+        if (isTypeColumnMissing == null) {
```

Comment:
Why not just setting `hasTypeColum` field in the `doImport` method when we retrieve the `headers` and avoid this method?